ftplugin/php.vim

```diff
@@ -1,58 +1,16 @@
-if exists('g:phpactorLoaded')
-  finish
-endif
-
-let g:phpactorLoaded = 1
-let g:phpactorpath = expand('<sfile>:p:h') . '/..'
-let g:phpactorbinpath = g:phpactorpath. '/bin/phpactor'
-let g:phpactorInitialCwd = getcwd()
-let g:phpactorCompleteLabelTruncateLength=50
-let g:_phpactorCompletionMeta = {}
-
-""
-" Path to the PHP binary used by Phpactor
-let g:phpactorPhpBin = get(g:, 'phpactorPhpBin', 'php')
-
-""
-" The Phpactor branch to use when calling @command(PhpactorUpdate)
-let g:phpactorBranch = get(g:, 'phpactorBranch', 'master')
-
-""
-" Automatically import classes when using VIM native omni-completion
-let g:phpactorOmniAutoClassImport = get(g:, 'phpactorOmniAutoClassImport', v:true)
-
-""
-" Ignore case when suggestion completion results
-let g:phpactorCompletionIgnoreCase = get(g:, 'phpactorCompletionIgnoreCase', 1)
-
-""
-" Function to use when populating a list of code references. The default
-" is to use the VIM quick-fix list.
-let g:phpactorQuickfixStrategy = get(g:, 'phpactorQuickfixStrategy', 'phpactor#quickfix#vim')
-
-""
-" Function to use when presenting a user with a choice of options. The default
-" is to use the VIM inputlist.
-let g:phpactorInputListStrategy = get(g:, 'phpactorInputListStrategy', 'phpactor#input#list#inputlist')
-
-""
-" When jumping to a file location: if the target file open in a window, switch
-" to that window instead of switching buffers.  The default is false.
-let g:phpactorUseOpenWindows = get(g:, 'phpactorUseOpenWindows', v:false)
-
-""
-" The list of files that determine workspace root directory
-" if contained within
-let g:phpactorProjectRootPatterns = get(g:, 'phpactorProjectRootPatterns', ['composer.json', '.git', '.phpactor.json', '.phpactor.yml'])
-
-""
-" The list of directories that should not be considered as workspace root directory
-" (in addition to '/' which is always considered)
-let g:phpactorGlobalRootPatterns = get(g:, 'phpactorGlobalRootPatterns', ['/', '/home'])
-
-if g:phpactorOmniAutoClassImport == v:true
-    autocmd CompleteDone *.php call phpactor#_completeImportClass(v:completed_item)
-endif
+function! s:import_on_complete_done(completed_item) abort
+    if get(b:, 'phpactorOmniAutoClassImport', g:phpactorOmniAutoClassImport) != v:true
+      return
+    endif
+    echom string(get(b:, 'phpactorOmniAutoClassImport', g:phpactorOmniAutoClassImport))
+
+    call phpactor#_completeImportClass(a:completed_item)
+endfunction
+
+augroup PhpactorInit
+    autocmd! * <buffer>
+    autocmd CompleteDone <buffer> call <SID>import_on_complete_done(v:completed_item)
+augroup END
```

Comment:
Both definitions will be still redefined each time similarly like those from before change.

Reply:
I did not test this behaviour yet. How it behaves with completion triggered by another client?

Reply:
> Both definitions will be still redefined each time similarly like those from before change.

I don't understand, which definition ? The auto commands ?
`autocmd! * <buffer>` will delete all auto commands from the group `PhpactorInit` linked to the current buffer.
That's the usual way of defining auto commands, you put them into a group so each time you want to define them, you first empty the all group and then declare them.
If they don't exists it does not throw an error, if they does then they are removed before being redefined.

Reply:
Maybe I wrote it not clearly enough. I did not mean what's going on inside this augroup. I meant redefinition of the whole `augroup PhpactorInit` and `import_on_complete_done()`. 

The function could be moved somewhere into autoload. 

I know and use the common way of defining augroups, however thanks for explanation.

It is usually not need to use `autocmd FileType` while its action can be simply used in ftplugin. Why not to move whole augroup to `plugin`and use 
```vim
autocmd FileType php autocmd CompleteDone <buffer=abuf> call <SID>import_on_complete_done(v:completed_item)
```
in this case? I still not tested the above proposition ;)

Reply:
Ok so for the first point: `import_on_complete_done` was removed in the other PR.
I was waiting for @dantleech opinion about integrating the logic into `phpactor#_importOnCompletedone` or whatever the name is :)

I don't mind the `augroup` being "redefine", it's suppose to work like that, there will still be only one group with this name.

I don't see how it will be beneficial to move the auto command in the `plugin` directory just to plug it on another auto command.
I guess it's because the file won't be source twice ?
But it does not matter, `ftplugin/php.vim` will also be loaded only once per buffer.
And it could be reloaded without any issues, it's not wrong to reload a file.
For instance I reload my vimrc constantly to update my plugins because they are not loaded by default. As long as the file is properly written it's not harmful.

Reply:
> I did not test this behaviour yet. How it behaves with completion triggered by another client?

Exactly as it has always done, there is an internal variable that is only filled by `phpactor#Complete`, so no matter how the method is called it will only work after calling `phpactor#Complete`.
This variable is reset after the function is done (we might want to add some try/catch in the mixt but not related to this PR) and on each call to `phpactor#Complete`.

Bottom line, the changes made here have no impact on the current behavior.

> Both definitions will be still redefined each time similarly like those from before change.

As explained it's not an issue for me, so if it's OK for you I'll resolve this comment soon.